src/worker/prettier-instance-worker.js

```diff
@@ -0,0 +1,108 @@
+  switch (type) {
+    case "import": {
+      const { modulePath } = payload;
+      let prettierInstance = path2ModuleCache.get(modulePath);
+      if (!prettierInstance) {
+        try {
+          prettierInstance = require(modulePath);
+          // If the instance is missing `format`, it's probably
+          // not an instance of Prettier
+          if (!prettierInstance.format) {
+            parentPort.postMessage({
+              type,
+              payload: { version: null },
+            });
+            break;
+          }
+          path2ModuleCache.set(modulePath, prettierInstance);
+        } catch {
+          parentPort.postMessage({
+            type,
+            payload: { version: null },
+          });
+          break;
+        }
+      }
+      parentPort.postMessage({
+        type,
+        payload: { version: prettierInstance.version },
+      });
+      break;
+    }
+    case "callMethod": {
+      const { modulePath, methodName, methodArgs, id } = payload;
+      const prettierInstance = path2ModuleCache.get(modulePath);
+      if (!prettierInstance) {
+        parentPort.postMessage({
+          type,
+          payload: {
+            result: new Error(
+              "Cannot load prettier instance from cache. Please call import() before call callMethod."
+            ),
+            id,
+            isError: true,
+          },
+        });
+        break;
+      }
+      let result;
+      try {
+        result = prettierInstance[methodName](...methodArgs);
+      } catch (error) {
+        parentPort.postMessage({
+          type,
+          payload: { result: error, id, isError: true },
+        });
+      }
+      if (result instanceof Promise) {
+        result.then(
+          (value) => {
+            try {
+              // For prettier-vscode, `languages` are enough
+              if (methodName === "getSupportInfo") {
+                value = { languages: value.languages };
+              }
+              parentPort.postMessage({
+                type,
+                payload: { result: value, id, isError: false },
+              });
+            } catch (error) {
+              parentPort.postMessage({
+                type,
+                payload: { result: error, id, isError: true },
+              });
+            }
+          },
+          (reason) => {
+            parentPort.postMessage({
+              type,
+              payload: { resuslt: reason, id, isError: true },
+            });
+          }
+        );
+        break;
+      }
+      try {
+        // For prettier-vscode, `languages` are enough
+        if (methodName === "getSupportInfo") {
+          result = { languages: result.languages };
+        }
+        parentPort.postMessage({
+          type,
+          payload: { result, id, isError: false },
+        });
+      } catch (error) {
+        parentPort.postMessage({
+          type,
+          payload: { result: error, id, isError: true },
```

Comment:
As I understant extra properties on `error` can't be cloned, for example parse error may have `codeFrame` assigned to error. But I'm not sure if we cares. 

https://github.com/prettier/prettier-synchronized/blob/97d9ba7abec8ea7f8de99ad182b6354f8739fa67/index.cjs#L44-L50

Reply:
Would it be better to serialize the error using serialize-error? https://www.npmjs.com/package/serialize-error

Reply:
serialize-error is a pure ESM package, so it is difficult to use from VSCode clients...

Reply:
Add simple serializer [901d664](https://github.com/prettier/prettier-vscode/pull/3016/commits/901d664801d8259f230eceb5e4db3723ab88791b)